eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -106,6 +106,10 @@ case class Commitments(channelVersion: ChannelVersion,
       remoteNextCommitInfo.left.toSeq.flatMap(_.nextRemoteCommit.spec.htlcs.collect(incoming).filter(expired).toSet)
   }
 
+  /**
+   * Find an outgoing HTLC that is signed by both parties in at least one of the current commitment transactions (local/remote).
```

Comment:
I find this confusing, we actually look for an HTLC that is signed by both parties in both (local and remote transactions). How about:

```suggestion
   * Find an outgoing HTLC that is present in both (local and remote) unrevoked commitment transactions. This means the HTLC is ready to be forwarded (it may already have been forwarded).
```

Reply:
> signed by both parties in both

But that's *not* what the function does, since it can find an HTLC that is signed only by us in the remote commitment (nextRemoteCommitInfo)...

Reply:
It wouldn't, otherwise `remoteSigned` would equal `None` and we would yield `None`.

Reply:
```scala
for {
 a <- Some(1)
 b <- Option.empty[Int]
} yield a
> val res0: Option[Int] = None
```

Reply:
Yes of course, but that's more subtle than that...
What you're implying is that if the HTLC is found in the `localCommit`, it will have been cross-signed in the `remoteCommit` before, right? And the reason we look into `remoteNextCommitInfo` is to explicitly NOT return it if the next commit fulfills/fails that HTLC?

Reply:
I'm not making assumptions on the order. Actually I think the function could be more precise because some cases (like the one you describe) are impossible. We just went with the simplest implementation.

I'm just saying that we check that the HTLC is present in both last valid local/remote transactions. If we are mid-failing, it's gonna only be in one transaction and the function won't return it.

Reply:
Clarified in https://github.com/ACINQ/eclair/pull/1578/commits/c4db58c5fcc6659669e122cc6afa9b8cebad6e6c